src/robusta/runner/config_loader.py

```diff
@@ -116,32 +120,81 @@ def __load_playbooks_repos(
-                        repo = GitRepo(playbooks_repo.url, playbooks_repo.key.get_secret_value(), playbooks_repo.branch)
-                        local_path = repo.repo_local_path
-                    elif playbooks_repo.url.startswith(LOCAL_PATH_URL_PREFIX):
-                        local_path = playbooks_repo.url.replace(LOCAL_PATH_URL_PREFIX, "")
-                    else:
-                        raise Exception(
-                            f"Illegal playbook repo url {playbooks_repo.url}. "
-                            f"Must start with '{GIT_SSH_PREFIX}', '{GIT_HTTPS_PREFIX}' or '{LOCAL_PATH_URL_PREFIX}'"
-                        )
-
-                    if not os.path.exists(local_path):  # in case the repo url was defined before it was actually loaded
-                        logging.error(f"Playbooks local path {local_path} does not exist. Skipping")
-                        continue
-
-                    # Adding to pip the playbooks repo from local_path
-                    subprocess.check_call([sys.executable, "-m", "pip", "install", "--no-build-isolation", local_path])
-                    playbook_package = self.__get_package_name(local_path=local_path)
+                    remove_pkg_path = None
+                    try:
+                        check_pkg_path = True
+                        if (
+                            (
+                                playbooks_repo.url.startswith("https://")
+                                or playbooks_repo.url.startswith("http://")
+                            ) and (
+                                playbooks_repo.url.endswith(".tar")
+                                or playbooks_repo.url.endswith(".tar.gz")
+                                or playbooks_repo.url.endswith(".tgz")
+                                or playbooks_repo.url.endswith(".tar.bz2")
+                                or playbooks_repo.url.endswith(".tbz2")
+                            )
+                        ):
+                            if playbooks_repo.url.startswith("http://"):
+                                logging.warning(
+                                    f"Downloading a playbook package from non-https source f{playbooks_repo.url}"
+                                )
+                            temp_dir = tempfile.mkdtemp()
+                            # after installing the downloaded package, remove it along with its directory
+                            remove_pkg_path = temp_dir
+                            pkg_name = urlparse(playbooks_repo.url).path.rsplit('/', 1)[-1]
+                            filename = os.path.join(temp_dir, pkg_name)
+                            self.download_file(playbooks_repo.url, filename, playbooks_repo.http_headers)
+                            pkg_path = f"file://{filename}"
+                            check_pkg_path = False
+                        elif playbooks_repo.url.startswith(GIT_SSH_PREFIX) or playbooks_repo.url.startswith(GIT_HTTPS_PREFIX):
```

Comment:
do these (below) also need the specified http_headers?

Reply:
Nope, it does not apply to them.